python/lance/__init__.py

```diff
@@ -68,15 +76,26 @@ def dataset(
     """
     if not filesystem:
         from pyarrow.fs import _resolve_filesystem_and_path
+
         filesystem, uri = _resolve_filesystem_and_path(uri, filesystem)
 
     if version is None:
-        if (
-            filesystem.get_file_info(os.path.join(uri, "_latest.manifest")).type
-            == pa.fs.FileType.NotFound
-        ):
+        if not _is_versioned(filesystem, uri):
             return _dataset_plain(uri, filesystem=filesystem, **kwargs)
 
+        if asof is not None:
+            ds = dataset(uri, version, filesystem=filesystem, **kwargs)
```

Comment:
use `_get_versioned_dataset` here? to reduce another I/O to check `is_plain_dataset` ?

Reply:
done